[ADD] Payroll: Work entry analysis report (#11063)
odoo/documentation: content/applications/hr/payroll/work_entry_analysis.rst

===================
Work entry analysis
===================

The default :guilabel:`Work entry analysis` report provides an overview of the validated work
entries for the current month. To view this report, navigate to :menuselection:`Payroll app -->
Reporting --> Work Entry Analysis`.

The work entries appear in a pivot table, with the default filters of :guilabel:`Current month:

> (Month)(Year)` and :guilabel:`Validated`. The various types of :doc:`work_entries` populate the rows
> (x-axis), while the :guilabel:`Total` values populate the only visible column (the y-axis).

Reviewer: ```suggestion
(Month)(Year)` and :guilabel:`Validated`. The various types of :doc:`work_entries` populate the rows,
while the :guilabel:`Total` values populate the only visible column.
```

Entry Type`, :guilabel:`Employee`, and :guilabel:`Department`. If in a multi-company database, a
:guilabel:`Company` option also appears.

Work entry analysis comparison
==============================

It is possible to compare the work entries from one time period to a previous time period. To view
this comparison, first navigate to :menuselection:`Payroll app --> Reporting --> Work Entry
Analysis`.

Next, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon in the search bar, revealing a
drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either
:guilabel:`Current Month: Previous Period` or :guilabel:`Current Month: Previous Year`.

The report updates and displays the data for the current time period, data for the selected previous
time period, as well as the :guilabel:`Variation` between the two, in a percentage.

.. image:: work_entry_analysis/work-entry-comparison.png
   :align: center
   :alt: A pivot table comparing the work entries of the current month and the previous month.

.. note::
   If no work entries for a specific :ref:`work entry type <payroll/work-entries>` are logged for

>    the time period, it does **not** appear on the report. That does **not** mean that the work entry

Reviewer: ```suggestion
   the time period, it does **not** appear on the report. That does **not** mean the work entry
```

>    type does not exist or is not configured.

Reviewer: ```suggestion
   type does not exist, or is not configured.
```

Use case: overtime report comparison
====================================

> It is possible to alter the *Work Entry Analysis* report to show a comparison of only overtime work

Reviewer: ```suggestion
It is possible to alter the *Work Entries Analysis* report to show a comparison of only overtime work
```

entries, grouped by employee, for a specific time period. To view this data, first navigate to the

> default *Work entry analysis* report by going to :menuselection:`Payroll app --> Reporting --> Work

Reviewer: ```suggestion
default *Work Entries Analysis* report by going to :menuselection:`Payroll app --> Reporting --> Work
```

Entry Analysis`.

Next, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon in the search bar, revealing a
drop-down menu. Under the :icon:`fa-filter` :guilabel:`Filters` column, click :guilabel:`Add Custom
Filter`, and a :guilabel:`Add Custom Filter` pop-up window appears.

Using the drop-down menu, select :guilabel:`Work Entry Type` for the first field, and select

> :guilabel:`Overtime Hours` for the last field. Click :guilabel:`Add`, and all other work entry types

Reviewer: might be nice to mention that the middle field ('is in') should stay the same

> disappear, and only :guilabel:`Overtime Hours` appears in the sole row.

Reviewer: ```suggestion
disappear, and :guilabel:`Overtime Hours` appear in the sole row.
```

To compare overtime from the current month to the previous month, to see which month had more
overtime logged, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon again in the search
bar. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click :guilabel:`Current Month:

> Previous Period`. Click away form the drop-down menu to close it.

Reviewer: ```suggestion
Previous Period`. Click away from the drop-down menu to close it.
```

> Now the report displays the :guilabel:`Overtime Hours` for the current month and the previous month,

Reviewer: ```suggestion
Now, the report displays the :guilabel:`Overtime Hours` for the current month and the previous month,
```

> along with the :guilabel:`variation`, in a percentage.

Reviewer: ```suggestion
along with the :guilabel:`Variation`, in a percentage.
```

To view which employees received the most overtime, click :icon:`fa-plus-square` :guilabel:`Overtime
Hours`, revealing a drop-down menu of options. Click :guilabel:`Employee`, and all employees with
overtime work entries for either the current or previous month appears.

In this example, it can be determined that :guilabel:`Marc Demo` worked the most overtime in

> :guilabel:`Agugust 2024`, whereas :guilabel:`Beth Evans` worked the most overtime hours in

Reviewer: ```suggestion
:guilabel:`August 2024`, whereas :guilabel:`Beth Evans` worked the most overtime hours in
```

the current month. To view this report, navigate to :menuselection:`Payroll app --> Reporting -->
Work Entry Analysis`.

The work entries appear in a pivot table, with the default filters of :guilabel:`Current month:
(Month)(Year)` and :guilabel:`Validated`. The various types of :doc:`work_entries` populate the
rows, while the :guilabel:`Total` values populate the only visible column.

To change the displayed information, click :icon:`fa-plus-square` :guilabel:`Total` above the main
column, revealing a drop-down menu of available metrics. Click on one of the available groupings,
and the data is further organized by that selected metric. The default options are :guilabel:`Work
Entry Type`, :guilabel:`Employee`, and :guilabel:`Department`. If in a multi-company database, a
:guilabel:`Company` option also appears.

Work entry analysis comparison
==============================

It is possible to compare the work entries from one time period to a previous time period. To view
this comparison, first navigate to :menuselection:`Payroll app --> Reporting --> Work Entry
Analysis`.

Next, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon in the search bar, revealing a

> drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either

Reviewer 2: Ah! It showed up after I selected a few months first:
![Screenshot 2024-10-08 at 4 00 54 PM](https://github.com/user-attachments/assets/08230337-30b1-4277-bb47-a26bd6154912)

It appears you need to select a time range first, in order to see comparison period options. Can you validate this and update the content to reflect the most accurate steps?

Also, it seems that in order to recreate your screenshot, the X-Axis group by needs to be explicitly declared: click **Total ➕ > Date created > Months**, no?

> drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either

Author: OK, I double checked on 3 runbots (all 17.0) and they all show me the comparison, by default:
![image](https://github.com/user-attachments/assets/8d3c2cdd-e235-4142-add1-0307569a045c)

The default filter is the current month- so if you don' have any information showing, that could be why it is not appearing. There has to be some data for it to show. Do you know if you had a blank/empty table when you didn't see it?

When I first was looking to see when the comparison appears, because I had issues in another report, you only need one specific date selected to make it appear. So if you were to remove ALL filters, and you select '2024', the comparison appears.  So as long as any time period is selected, whether it's a month, year, or quarter then it appears - since it will compare the data with the previous of whatever is selected (quarter, month, etc).

> drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either

Author: I also double-checked the X-axis question, and the default displays it like in my screenshot, but if you select something else or make changes to the default, it might not appear like that. So I think I can either add a note that states that, and that should clear things up!

> drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either

Author: OK, the note was updated to be more clear. LMK if you still have any issues- but you should be good!

> drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either

Reviewer 2: I just confirmed that if you follow the instructions on line 59 exactly, and _don't_ remove the default filters when landing on the dashboard (like I did earlier), then, indeed, the comparison options show up.

It does seem like it won't show up unless a time window is explicitly selected. Maybe worth adding as a **Note**, yea...since removing default filters is almost a knee-jerk reaction for anyone who regularly runs reports (you instead just go right to custom and build your own logic, usually)

column, revealing a drop-down menu of available metrics. Click on one of the available groupings,
and the data is further organized by that selected metric. The default options are :guilabel:`Work
Entry Type`, :guilabel:`Employee`, and :guilabel:`Department`. If in a multi-company database, a
:guilabel:`Company` option also appears.

Work entry analysis comparison
==============================

It is possible to compare the work entries from one time period to a previous time period. To view
this comparison, first navigate to :menuselection:`Payroll app --> Reporting --> Work Entry
Analysis`.

Next, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon in the search bar, revealing a
drop-down menu. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click on either
:guilabel:`Current Month: Previous Period` or :guilabel:`Current Month: Previous Year`.

The report updates and displays the data for the current time period, data for the selected previous
time period, as well as the :guilabel:`Variation` between the two, in a percentage.

.. image:: work_entry_analysis/work-entry-comparison.png

>    :align: center

Reviewer 3: Please remove `:align:` attribute :)
```suggestion
```

overtime logged, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` icon again in the search
bar. Under the :icon:`fa-adjust` :guilabel:`Comparison` section, click :guilabel:`Current Month:
Previous Period`. Click away from the drop-down menu to close it.

Now, the report displays the :guilabel:`Overtime Hours` for the current month and the previous
month, along with the :guilabel:`Variation`, in a percentage.

To view which employees received the most overtime, click :icon:`fa-plus-square` :guilabel:`Overtime
Hours`, revealing a drop-down menu of options. Click :guilabel:`Employee`, and all employees with
overtime work entries for either the current or previous month appears.

In this example, it can be determined that :guilabel:`Marc Demo` worked the most overtime in
:guilabel:`August 2024`, whereas :guilabel:`Beth Evans` worked the most overtime hours in
:guilabel:`September 2024`. Additionally, :guilabel:`Mitchell Admin` had the largest variation
change, with a :guilabel:`-100%` change from :guilabel:`August 2024` to :guilabel:`September 2024`.

.. image:: work_entry_analysis/variation.png

>    :align: center

Reviewer 3: ```suggestion
```
